### PR TITLE
feat(cli): adding option to compile as dll

### PR DESCRIPTION
**Pull request by Alvarwow69**
Add new option that allow you to build your payload as a `.dll`

### CODE REVIEW

File: cli/parser.go [hunk 1]
```diff
@@ -42,6 +42,26 @@ const ASCII_ART = `
 
     `
 
+func BuildLoader(opts *Options) *exec.Cmd {
+	os.Setenv("GOOS", opts.OS)
+	os.Setenv("GOARCH", opts.Arch)
+	os.Setenv("CGO_ENABLED", "1")
```
Comment (djnnvx) [inline cli/parser.go]: can you only enable this for dll compilation please ?

File: cli/parser.go [hunk 2]
```diff
@@ -42,6 +42,26 @@ const ASCII_ART = `
...
+	os.Setenv("CC", "x86_64-w64-mingw32-gcc")
```
Comment (djnnvx) [inline cli/parser.go]: same for this, please

File: cli/parser.go [hunk 3]
```diff
@@ -42,6 +42,26 @@ const ASCII_ART = `
...
+
+	if opts.BuildType == "dll" {
+		fmt.Println("[*] Compile payload as dll...")
+
+		return exec.Command("go", "build", "-buildmode=c-shared", "-ldflags", "-s -w -H=windowsgui", "-o", "payload.dll", ".")
+	} else if opts.BuildType == "exe" {
+		fmt.Println("[*] Compile payload as executable...")
```
Comment (djnnvx) [inline cli/parser.go]: should be `Compiling`

File: cli/parser.go [hunk 4]
```diff
@@ -42,6 +42,26 @@ const ASCII_ART = `
...
     `
 
+func BuildLoader(opts *Options) *exec.Cmd {
+	os.Setenv("GOOS", opts.OS)
+	os.Setenv("GOARCH", opts.Arch)
+	os.Setenv("CGO_ENABLED", "1")
+	os.Setenv("CC", "x86_64-w64-mingw32-gcc")
+
+	if opts.BuildType == "dll" {
+		fmt.Println("[*] Compile payload as dll...")
```
Comment (djnnvx) [inline cli/parser.go]: should be `Compiling`

File: tools/utils.go [hunk 5]
```diff
@@ -132,27 +132,44 @@ func GetMainTemplate(
 	key string,
 	sc string,
 	sleepTime uint,
+	export bool,
```
Comment (djnnvx) [inline tools/utils.go]: boolean should have a verb inside the variable name

in this case, i think `shouldExport` is good

File: tools/utils.go [hunk 6]
```diff
@@ -132,27 +132,44 @@ func GetMainTemplate(
...
+%s
 
-func main() {
+	f, err := os.Create("./data.txt")
+	defer f.Close()
+    _, err2 := f.WriteString("old falcon\n")
+	if err2 != nil {
+        fmt.Println("c'est fait ça marche pas mais c'est quand même exec tqt c'est trop bien")
+    }
+	fmt.Println("c'est fait ça marche c'est trop bien")
```
Comment (djnnvx) [inline tools/utils.go]: can you remove this test-code please ?

**Review by djnnvx — CHANGES_REQUESTED** [review]
good work, some stuff is still missing though

dont forget to run `gofmt -w .` to make sure all is formatted properly, thanks

**Comment by Alvarwow69** [comment]
All good

**Review by djnnvx — APPROVED** [review]
good work

**Pull request merged**